apps/cowswap-frontend/src/cowSdk.ts

```diff
@@ -1,9 +1,16 @@
 import { MetadataApi } from '@cowprotocol/app-data'
-import { OrderBookApi } from '@cowprotocol/cow-sdk'
+import { ORDER_BOOK_PROD_CONFIG, ORDER_BOOK_STAGING_CONFIG, OrderBookApi } from '@cowprotocol/cow-sdk'
 
 import { isBarnBackendEnv } from 'legacy/utils/environments'
 
+const prodBaseUrls = process.env.REACT_APP_ORDER_BOOK_URLS
+  ? JSON.parse(process.env.REACT_APP_ORDER_BOOK_URLS)
```

Comment:
Should it handle the case where the contents is not a valid JSON?

Reply:
In this case, the app just wouldn't run and we definitely will notice it

Reply:
Hmm more or less.
The error is noticeable alright

![image](https://github.com/cowprotocol/cowswap/assets/43217/d870c245-9f23-405a-97e5-cca8613d8a23)

The problem is this is thrown only at run time. The app does not fail to start when I pass a bad value:
![image](https://github.com/cowprotocol/cowswap/assets/43217/0cabd44c-c3aa-4e76-8474-32a38e7e6d6a)

So you are right, no need to handle it here, but we should fetch and fail sooner, probably at build time.

Reply:
`REACT_APP_ORDER_BOOK_URLS` env var is supposed to be used only for local testing purposes, it should never be used in prod. I don't think we should fail build because of it.
For example, when `REACT_APP_PINATA_API_KEY` is not set or contains invalid value we don't fail build.